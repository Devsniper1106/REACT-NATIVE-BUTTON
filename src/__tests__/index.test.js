import React from "react";
import renderer from "react-test-renderer";
import AwesomeButton from "../index";

const DEFAULT_TEXT = "Default";

describe("AwesomeButton", () => {
  it("should render", () => {
    const component = renderer.create(<AwesomeButton />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it("should render with a view container", () => {
    const component = renderer.create(<AwesomeButton />).toJSON();

    expect(component.type).toEqual("View");
  });

  it("should render a default text children", () => {
    const component = renderer.create(
      <AwesomeButton>{DEFAULT_TEXT}</AwesomeButton>
    );
    const element = component.root.findByProps({
      testID: "aws-btn-content-text"
    });

    expect(element.props.children).toEqual(DEFAULT_TEXT);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should render the button shadow element", () => {
    const component = renderer.create(
      <AwesomeButton>{DEFAULT_TEXT}</AwesomeButton>
    );
    const { height, raiseLevel } = component.root.props;
    const element = component.root.findByProps({ testID: "aws-btn-shadow" });

    expect(element.props.style[1].height).toBe(height - raiseLevel);
    expect(element.props.testID).toBe("aws-btn-shadow");
  });

  it("should render the button bottom element", () => {
    const component = renderer.create(
      <AwesomeButton>{DEFAULT_TEXT}</AwesomeButton>
    );
    const { height, raiseLevel } = component.root.props;
    const element = component.root.findByProps({ testID: "aws-btn-bottom" });

    expect(element.props.style[1].height).toBe(height - raiseLevel);
    expect(element.props.testID).toBe("aws-btn-bottom");
  });

  it("should render the button content element", () => {
    const component = renderer.create(
      <AwesomeButton>{DEFAULT_TEXT}</AwesomeButton>
    );
    const { height, raiseLevel } = component.root.props;
    const element = component.root.findByProps({ testID: "aws-btn-content" });

    expect(element.props.style[1].height).toBe(height - raiseLevel);
    expect(element.props.testID).toBe("aws-btn-content");
  });

  it("should render the button text view", () => {
    const component = renderer.create(
      <AwesomeButton>{DEFAULT_TEXT}</AwesomeButton>
    );
    const element = component.root.findByProps({ testID: "aws-btn-text" });

    expect(element.props.testID).toBe("aws-btn-text");
  });

  it("should render the button progress view", () => {
    const component = renderer.create(
      <AwesomeButton progress>{DEFAULT_TEXT}</AwesomeButton>
    );
    const view = component.root.findByProps({ testID: "aws-btn-content-view" });
    // const element = component.root.findByProps({ testID: "aws-btn-progress" });
    expect(view.props.testID).toBe("aws-btn-content-view");
  });

  it("should render the button active background", () => {
    const component = renderer.create(
      <AwesomeButton>{DEFAULT_TEXT}</AwesomeButton>
    );
    const element = component.root.findByProps({
      testID: "aws-btn-active-background"
    });

    expect(element.props.testID).toBe("aws-btn-active-background");
  });

  it("should render the content placeholder on empty button", () => {
    const component = renderer.create(<AwesomeButton />);
    const element = component.root.findByProps({
      testID: "aws-btn-content-placeholder"
    });

    expect(element.props.testID).toBe("aws-btn-content-placeholder");
  });
});
