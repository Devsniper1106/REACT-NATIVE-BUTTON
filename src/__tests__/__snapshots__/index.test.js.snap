// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwesomeButton should render 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 60,
      "opacity": 0,
      "width": null,
      "zIndex": 10,
    }
  }
  testID="aws-btn-content-view"
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.15)",
        "borderBottomLeftRadius": null,
        "borderBottomRightRadius": null,
        "borderRadius": 4,
        "borderTopLeftRadius": null,
        "borderTopRightRadius": null,
        "bottom": -2,
        "height": 56,
        "left": "1%",
        "position": "absolute",
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
        "width": "98%",
      }
    }
    testID="aws-btn-shadow"
  />
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
        },
        Object {
          "backgroundColor": "#9f9f9f",
          "borderBottomLeftRadius": null,
          "borderBottomRightRadius": null,
          "borderRadius": 4,
          "borderTopLeftRadius": null,
          "borderTopRightRadius": null,
          "height": 56,
          "width": null,
        },
      ]
    }
    testID="aws-btn-bottom"
  />
  <View
    style={
      Object {
        "borderBottomLeftRadius": null,
        "borderBottomRightRadius": null,
        "borderRadius": 4,
        "borderTopLeftRadius": null,
        "borderTopRightRadius": null,
        "height": 56,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
        "width": null,
      }
    }
    testID="aws-btn-content"
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "overflow": "hidden",
            "zIndex": 3,
          },
          Object {
            "backgroundColor": "#c0c0c0",
            "borderBottomLeftRadius": null,
            "borderBottomRightRadius": null,
            "borderColor": null,
            "borderRadius": 4,
            "borderTopLeftRadius": null,
            "borderTopRightRadius": null,
            "borderWidth": 0,
          },
        ]
      }
      testID="aws-btn-text"
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.15)",
            "height": 56,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "width": null,
          }
        }
        testID="aws-btn-active-background"
      />
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.15)",
            "height": 20,
            "opacity": 1,
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": "55%",
          }
        }
        testID="aws-btn-content-placeholder"
      />
    </View>
  </View>
</View>
`;

exports[`AwesomeButton should render a default text children 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 60,
      "opacity": 0,
      "width": null,
      "zIndex": 10,
    }
  }
  testID="aws-btn-content-view"
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.15)",
        "borderBottomLeftRadius": null,
        "borderBottomRightRadius": null,
        "borderRadius": 4,
        "borderTopLeftRadius": null,
        "borderTopRightRadius": null,
        "bottom": -2,
        "height": 56,
        "left": "1%",
        "position": "absolute",
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
        "width": "98%",
      }
    }
    testID="aws-btn-shadow"
  />
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
        },
        Object {
          "backgroundColor": "#9f9f9f",
          "borderBottomLeftRadius": null,
          "borderBottomRightRadius": null,
          "borderRadius": 4,
          "borderTopLeftRadius": null,
          "borderTopRightRadius": null,
          "height": 56,
          "width": null,
        },
      ]
    }
    testID="aws-btn-bottom"
  />
  <View
    style={
      Object {
        "borderBottomLeftRadius": null,
        "borderBottomRightRadius": null,
        "borderRadius": 4,
        "borderTopLeftRadius": null,
        "borderTopRightRadius": null,
        "height": 56,
        "left": 0,
        "position": "absolute",
        "top": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
        "width": null,
      }
    }
    testID="aws-btn-content"
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "overflow": "hidden",
            "zIndex": 3,
          },
          Object {
            "backgroundColor": "#c0c0c0",
            "borderBottomLeftRadius": null,
            "borderBottomRightRadius": null,
            "borderColor": null,
            "borderRadius": 4,
            "borderTopLeftRadius": null,
            "borderTopRightRadius": null,
            "borderWidth": 0,
          },
        ]
      }
      testID="aws-btn-text"
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.15)",
            "height": 56,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "width": null,
          }
        }
        testID="aws-btn-active-background"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": null,
            "fontSize": 14,
            "fontWeight": "bold",
            "opacity": 1,
            "paddingBottom": 0,
            "paddingHorizontal": 12,
            "paddingTop": 0,
            "textAlign": "center",
            "transform": Array [
              Object {
                "scale": 1,
              },
            ],
            "width": "100%",
            "zIndex": 10,
          }
        }
        testID="aws-btn-content-text"
      >
        Default
      </Text>
    </View>
  </View>
</View>
`;
